2.0/m_apacheauth.cpp

```diff
@@ -0,0 +1,168 @@
+/*       +------------------------------------+
+ *       | Inspire Internet Relay Chat Daemon |
+ *       +------------------------------------+
+ *
+ *  InspIRCd: (C) 2002-2010 InspIRCd Development Team
+ * See: http://wiki.inspircd.org/Credits
+ *
+ * This program is free but copyrighted software; see
+ *            the file COPYING for details.
+ *
+ * ---------------------------------------------------
+ */
+
+#include "inspircd.h"
+#include "hash.h"
+
+/* $ModDesc: Allow/Deny connections based upon an apache auth file */
+/* $LinkerFlags: -lcrypt */
+
+class ModuleApacheAuth : public Module
+{
+	std::string authfile;
+	std::string killreason;
+	std::string allowpattern;
+	bool verbose;
+	std::map<std::string, std::string> logins;
+
+ public:
+	ModuleApacheAuth()
+	{
+	}
+
+	void init()
+	{
+		OnRehash(NULL);
+		Implementation eventlist[] = { I_OnRehash, I_OnUserRegister };
+		ServerInstance->Modules->Attach(eventlist, this, 2);
+	}
+
+	void OnRehash(User* user)
+	{
+		ConfigTag* conf = ServerInstance->Config->ConfValue("apacheauth");
+		authfile = conf->getString("authfile");
+		killreason = conf->getString("killreason");
+		allowpattern = conf->getString("allowpattern");
+		verbose = conf->getBool("verbose");
+		LoadAuthFile();
+	}
+
+	bool SplitRecord(const std::string &record, std::string &record_algorithm, std::string &record_salt, std::string &record_hash)
+	{
+		const char *s = record.c_str();
+		const char *ptr;
+
+		if (record[0]!='$') return false;
+		s = s+1;
+		ptr = strchr(s, '$');
```

Comment:
You can use std::string::find here instead of using `const char*` and `strchr` here.